Ludown/lib/parseFileContents.js

```diff
@@ -567,6 +543,10 @@ const parseAndHandleListEntity = function(parsedContent, chunkSplitByLine) {
                 if(!subListExists[0].list.includes(listItem)) subListExists[0].list.push(listItem);
             })
         }
+        // see if the roles all exist and if not, add them
+        entityRoles.forEach(role => {
+            if(!closedListExists[0].roles.includes(role)) closedListExists[0].roles.push(role);
```

Comment:
Same here. `includes()` is technically a loop. Consider using a map of roles:
```js
const rolesMap = closedListExists[0].roles.reduce((map, role) => (map[role] = true, map), {});
entityRoles.forEach(role => {
  if (!rolesMap[role]) {
    closedListExists[0].roles.push(role);
  }
});
```
This will reduce `O(n^2)` to `O(n * 2)`